odoo/modules/loading.py

```diff
@@ -61,21 +60,15 @@ def _get_files_of_kind(kind):
         return files
 
     filename = None
-    try:
-        if kind in ('demo', 'test'):
-            threading.current_thread().testing = True
-        for filename in _get_files_of_kind(kind):
-            _logger.info("loading %s/%s", package.name, filename)
-            noupdate = False
-            if kind in ('demo', 'demo_xml') or (filename.endswith('.csv') and kind in ('init', 'init_xml')):
-                noupdate = True
-            tools.convert_file(env, package.name, filename, idref, mode, noupdate, kind)
-    finally:
-        if kind in ('demo', 'test'):
-            threading.current_thread().testing = False
-
+    for filename in _get_files_of_kind(kind):
```

Comment:
Actually this changes implies some more questions: all logic previously protected by `threading.current_thread().testing `  may be enabled during demo data install. One of the implication is potential commits, send mail, .... If you commit during demo it may work, but be a problem in case of failure. In this case demo data should be roll-backed but the install should be able to continue.
We may consider patching the cursor to avoid a commit during demo data install.

Reply:
> Patching is also not thread-safe. 

The cursor used for the install is only available in the current_thread, this would be the cursor to patch. Could be just a log_error if committed. Again, unsure but doesn't cost a lot to check that.

> You could have the same failures during installation of normal data for modules.

Yes this is True, but the behaviour is not exactly the same. For demo data the install will continue for data it will stop. But it is True that committing during data is not a good idea either since you could have a partially installed module maybe in a broken state.

> Installing demo data should have a different logic than testing. I would also argue that if you could commit during installation of demo data, that data is correctly loaded and the remaining incorrect data will be rolled back, so this should not be an issue - it is for demo purposes anyway.

Possible, but the testing flag was added there for a reason. The logic is different than testing yes but since the flag was there historically, we need to check that it is not a problem to remove it ;).

Reply:
`current_test` which is used to check if we are testing is only available globally, not per cursor/thread.

I'm checking the e-mails first and I'll check if there are some reasons in commits for this.

Reply:
Only task-1856493 is mentioned in the history: it is about loading demo data so that users can test Odoo - nothing related to actually running the test framework and no reason to why demo data could be loaded in a testing context.